File: mongodb-service/src/models/clusterModel.ts
import * as mongoose from "mongoose";
import { Model } from "mongoose";

/**
 * Cluster type, model mongoose document and schema definition. Models are used by
 * Controllers so they are picked up by the swagger. Documents and schemas are used by 
 * Repositories so model defined Mongodb functions can be used.
 */
type ClusterType = ICluster & mongoose.Document;
export interface IClusterModel {
    _id?: string;
    aggId?: string;
    aggName: string;
    identifier: string;
    identifierType: string;
    xAxis: string;
    xType: string;
    yAxis: string;
    yType: string;
}

export interface ICluster extends IClusterModel, mongoose.Document {
    _id: string;
}

export const ClusterSchema = new mongoose.Schema({
    aggId: {
        required: true,
        type: String
    },
    aggName: {
        required: true,
        type: String
    },
    identifier: {
        required: true,
        type: String
    },
    identifierType: {
        required: true,
        type: String
    },
    xAxis: {
        required: true,
        type: String
    },
    xType: {
        required: true,
        type: String
    },
    yAxis: {
        required: true,
        type: String
    },
    yType: {
        required: true,
        type: String
    }
});

const Cluster: Model<ICluster> = mongoose.model<ClusterType>("Cluster", ClusterSchema);
export default Cluster;

File: analytic-engine/src/main/java/com/mycompany/models/ClusterModel.java
package com.mycompany.models;

import java.util.Objects;

public class ClusterModel {
    public String _id;
    public String __v;
    public String aggId;
    public String aggName;
    public String identifier;
    public String identifierType;
    public String xAxis;
    public String xType;
    public String yAxis;
    public String yType;

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        ClusterModel that = (ClusterModel) o;
        return Objects.equals(_id, that._id) &&
                Objects.equals(__v, that.__v) &&
                Objects.equals(aggId, that.aggId) &&
                Objects.equals(aggName, that.aggName) &&
                Objects.equals(identifier, that.identifier) &&
                Objects.equals(identifierType, that.identifierType) &&
                Objects.equals(xAxis, that.xAxis) &&
                Objects.equals(xType, that.xType) &&
                Objects.equals(yAxis, that.yAxis) &&
                Objects.equals(yType, that.yType);
    }
}
